stratum/hal/bin/np4intel/np4intel.bzl

```diff
@@ -0,0 +1,37 @@
+# This Skylark rule imports the netcope shared libraries and headers since 
+# there is not yet any native support for Bazel in netcope. The NP4_INSTALL
+# environment variable needs to be set, otherwise the Stratum rules which
+# depend on netcope cannot be built.
+
+
+def _impl(repository_ctx):
+    if "NP4_INSTALL" not in repository_ctx.os.environ:
+        repository_ctx.file("BUILD", """
+""")
+        return
+    netcope_path = repository_ctx.os.environ["NP4_INSTALL"]
+    repository_ctx.symlink(netcope_path, "netcope-bin")
+    repository_ctx.file("BUILD", """
+package(
+    default_visibility = ["//visibility:public"],
+)
+# trick to export headers in a convenient way
+cc_library(
+    name = "np4_headers",
+    hdrs = glob(["netcope-bin/include/np4/*.hpp", 
+                 "netcope-bin/include/np4/p4*.h"]),
+    includes = ["netcope-bin/include"],
+)
+
+cc_import(
+  name = "np4_atom",
+  hdrs = [], # see cc_library rule above 
+  shared_library = "netcope-bin/lib/libnp4atom.so",
+)
+
+""")
+
+np4intel_configure = repository_rule(
+    implementation=_impl,
+    local = True,
+    environ = ["NP4_INSTALL"])
```

Comment:
Convert to proper dependency, if possible.
Example for external SDKs: https://github.com/stratum/stratum/blob/07d6e325e3da6694bf3eb04e116d61630555d42e/bazel/deps.bzl#L272-L288

Reply:
To include local files, `urls` supports the `file://` scheme:

```
        http_archive(
            name = "some_dpe",
            sha256 = "5aaeac84cbc6f1869c2ad99a973e107c67e0f99bb8c5c4e02282f205c2b3acf9",
            urls = ["file:///tmp/file.tar"],
            build_file = "@//bazel:external/some_dep.BUILD",
        )
```

Reply:
Hi Max, I think I mentioned this in a previous TST meeting, I can't include the netcope tarball in the bazel build dependency as the tarball is actually installed as part of the docker build process by running a shell script supplied by Netcope in the tarball directory (see here https://github.com/craigsdell/stratum/blob/dev/netcope/stratum/hal/bin/np4intel/docker/Dockerfile.builder).  I guess we could put a checksum test in the docker build file, but right now Netcope don't have a standard tarball that they supply so we might need to approach them about that so that it's a consistent tarball supplied to all customers/users of stratum and we could include a checksum test in the docker build.

Reply:
The tarball was just an example, if you're current using an unpacked directory, [new_local_repository](https://docs.bazel.build/versions/master/be/workspace.html#new_local_repository) would be the right choice.